source/core/Array.cpp

```diff
@@ -1121,32 +1121,58 @@ VIREO_FUNCTION_SIGNATURET(ArrayMaxMinInternal, FindArrayMaxMinInstruction)
     AQBlock1* maxValue = minValue;
     if (len == 0)
         maxIndex = minIndex = -1;
+
+    Boolean haveSeenPositiveInfinity = false;
+    Boolean haveSeenNegativeInfinity = false;
     for (IntIndex i = 0; i < len; i++) {
-        Boolean less;
-        snippet->_p0 = arrayIn->BeginAt(i);
+        Boolean shouldUpdateMaxOrMin = false;
+        AQBlock1* currentElement = arrayIn->BeginAt(i);
+        Double currentElementValue = *((Double*)currentElement);
+        Boolean isANumber = !::isnan(currentElementValue);
+        Boolean isPositiveInfinity = ::isinf(currentElementValue) && currentElementValue > 0;
+        Boolean isNegativeInfinity = ::isinf(currentElementValue) && currentElementValue < 0;
```

Comment:
I don't think you even need to check for +/- infinity, either.  The normal comparison will already handle these cases.